### PR TITLE
Feat hexa refactor

### PR DESCRIPTION
**Pull request by khlam**
# Description of changes:
- Curseforge now parses Owners/Contributors into "authors"
- Refactor bugs fixed
- Up-to-date with master

# Action List:
- [x] I've bumped `package.json` version (minor or major.)
- [x] I've kept this branch up-to-date with `dev`.

### CODE REVIEW

File: src/checkWhichHost/index.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export function checkWhichHost (URL) {
+  if (URL.startsWith('https://www.curseforge.com/wow/addons/')) {
+    return initCurseforgeObj(URL)
+  }
+
+  const errorObj = { 'error': "ERROR: Invalid URL '" + URL + "'. Given link does not match parse" }
+  return errorObj
+}
+
+function initCurseforgeObj (URL) {
```
Comment (khlam) [inline src/checkWhichHost/index.js]: Same function

File: src/pageParserAdapter/curseforge.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import { XMLHttpRequest } from 'xmlhttprequest'
+import { JSDOM } from 'jsdom'
+
+export function parseUrl (url) {
```
Comment (khlam) [inline src/pageParserAdapter/curseforge.js]: same function

File: src/checkWhichHost/index.js (hunk 1 above)
Comment (zainkai) [reply src/checkWhichHost/index.js]: what do you mean?

Comment (khlam) [reply src/checkWhichHost/index.js]: @zainkai Referring to how this function and the other one I marked are pretty much exactly the same function, but only one is called, I'll leave it up to you how you want to organize how they're called/what to keep

Comment (zainkai) [reply src/checkWhichHost/index.js]: If that's the case we should be using parseUrl. We might want to look into creating an addon class to keep track of properties. But for the time being replace initCurseforgeObj with parseUrl and we can do those changes later down the line.

**Pull request merged**